alternator/executor.cc

```diff
@@ -4794,11 +4794,9 @@ future<executor::request_return_type> executor::describe_endpoints(client_state&
 
 static std::map<sstring, sstring> get_network_topology_options(service::storage_proxy& sp, gms::gossiper& gossiper, int rf) {
     std::map<sstring, sstring> options;
-    sstring rf_str = std::to_string(rf);
-    auto& topology = sp.get_token_metadata_ptr()->get_topology();
-    for (const gms::inet_address& addr : gossiper.get_live_members()) {
-        options.emplace(topology.get_datacenter(addr), rf_str);
-    };
+    for (const auto& dc : sp.get_token_metadata_ptr()->get_topology().get_datacenters()) {
+        options.emplace(dc, std::to_string(rf));
```

Comment:
I'd keep rf_str around just to make debugging easier. Imagine you're debugging this, you will step into to_string() all the time instead of options.emplace(). Not a blocker for a merge I guess.

Reply:
It will probably be optimized out anyway.

Reply:
I'm not convinced by the debugability argument - I think it makes the code uglier and nobody will ever gdb this code - but I agree it makes sense to do the integer to string conversion just once instead of in the loop to make it clearer to the reader what happens after the optimizer takes its axe to this code.
But I think the code is also fine as it is so I'm "approve"ing.